test/unit/action/autopilot.spec.js

```diff
@@ -59,4 +70,97 @@ describe('Action Autopilot Unit Test', () => {
+        scores: [
+          {
+            alias: 'some-alias',
+            public_key: 'some-pubkey',
+            score: 14035087,
+          },
+        ],
+      };
+      info.getInfo.resolves();
+      store.network = 'testnet';
+    });
+
+    it('should fail if network cannot be read', async () => {
+      store.network = null;
+      await autopilot.updateNodeScores();
+      expect(logger.error, 'was called once');
+      expect(grpc.sendAutopilotCommand, 'was not called');
+      expect(db.save, 'was not called');
+      expect(store.settings.nodeScores, 'to equal', {});
+    });
+
+    it('should read scores for testnet from empty cache', async () => {
+      nock('https://nodes.lightning.computer')
+        .get('/availability/v1/btctestnet.json')
+        .reply(500, 'Boom!');
+
+      await autopilot.updateNodeScores();
+      expect(grpc.sendAutopilotCommand, 'was not called');
+      expect(logger.error, 'was called twice');
+      expect(db.save, 'was not called');
+      expect(store.settings.nodeScores, 'to equal', {});
+    });
+
+    it('should read scores for testnet from cache', async () => {
+      nock('https://nodes.lightning.computer')
+        .get('/availability/v1/btctestnet.json')
+        .reply(500, 'Boom!');
+      store.settings.nodeScores = {
+        testnet: { 'some-pubkey': 0.14035087 },
+      };
+
+      await autopilot.updateNodeScores();
+      expect(grpc.sendAutopilotCommand, 'was called with', 'setScores', {
+        heuristic: 'externalscore',
+        scores: { 'some-pubkey': 0.14035087 },
+      });
+      expect(logger.error, 'was called once');
+      expect(db.save, 'was not called');
+      expect(store.settings.nodeScores, 'to equal', {
+        testnet: { 'some-pubkey': 0.14035087 },
+      });
+    });
+
+    it('should set scores for testnet', async () => {
+      nock('https://nodes.lightning.computer')
+        .get('/availability/v1/btctestnet.json')
+        .reply(200, scoresJson);
+
+      await autopilot.updateNodeScores();
+      expect(grpc.sendAutopilotCommand, 'was called with', 'setScores', {
+        heuristic: 'externalscore',
+        scores: { 'some-pubkey': 0.14035087 },
+      });
+      expect(db.save, 'was called once');
+      expect(store.settings.nodeScores, 'to equal', {
+        testnet: { 'some-pubkey': 0.14035087 },
+      });
+    });
+
+    it('should set scores for mainnet', async () => {
+      store.network = 'mainnet';
+      nock('https://nodes.lightning.computer')
+        .get('/availability/v1/btc.json')
+        .reply(200, scoresJson);
+
+      await autopilot.updateNodeScores();
+      expect(grpc.sendAutopilotCommand, 'was called with', 'setScores', {
+        heuristic: 'externalscore',
+        scores: { 'some-pubkey': 0.14035087 },
+      });
+      expect(db.save, 'was called once');
+      expect(store.settings.nodeScores, 'to equal', {
+        mainnet: { 'some-pubkey': 0.14035087 },
+      });
+    });
+  });
```

Comment:
We should also test all relevant code branches much more thoroughly. I rewrote the unit tests from scratch.